app/models/appointment.rb

```diff
@@ -0,0 +1,21 @@
+class Appointment < ActiveRecord::Base
+  before_save :increment_version, :create_version, if: :changed?
+
+  validates :uid, :booked_at, :booking_at, :booking_status, :delivery_partner,
+            presence: true
+  validates :uid, uniqueness: true
+  validates :delivery_partner, inclusion: { in: Partners.delivery_partners }
+
+  scope :bookings, ->(end_point) { where('booked_at <= ?', end_point).where(cancelled: false) }
+  scope :completed_transactions, ->(end_point) { transactions(end_point).where(booking_status: 'Complete') }
```

Comment:
Are we going to keep the `transaction` language around now we've renamed elsewhere?